collated/main/A0141128R.md

```diff
@@ -0,0 +1,692 @@
+            + "Parameters: <index> <details> by/on <date> at <time> /<priority> /<TAG...>\n"
+    		+ "NOTE: You must reenter all parameters again.\n"
+            + "Example: " + COMMAND_WORD + " 2 Take Bongo out for a walk tomorrow 2pm /medium -dog";
+    
+    public static final String MESSAGE_EDITED_TASK_SUCCESS = "Edited task.";
+
+    private final int targetIndex;
+    private final Task toEditWith;
+    private final boolean floating;
+    private final boolean removePriority;
+    private boolean tagIsEmpty = false;
+    
+    
+    public EditCommand(	boolean removePriority,
+    					boolean floating, 
+    					int targetIndex,
+    					String newDetail,
+    					LocalDate newDueByDate,
+    					LocalTime newDueByTime,
+    					String newPriority,
+    					Set<String> newTags) throws IllegalValueException {
+        final Set<Tag> tagSet = new HashSet<>();
+        if(newTags.isEmpty())
+        	tagIsEmpty = true;
+        //System.out.println(tagIsEmpty);
+        for (String tagName : newTags) {
+            tagSet.add(new Tag(tagName));
+        }
+        this.toEditWith = new Task(
+                new Detail(newDetail),
+                new DueByDate (newDueByDate),
+                new DueByTime(newDueByTime),
+                new Priority(newPriority),
+                new UniqueTagList(tagSet)
+        );
+        this.floating = floating;
+        this.targetIndex = targetIndex;
+        this.removePriority = removePriority;
+    }
+        
+    /**
+     * For RANGE DATE AND TIME
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     * 
+```
+###### /java/seedu/cmdo/logic/commands/EditCommand.java
+``` java
+     */
+    public EditCommand(boolean removePriority, 
+    				  int targetIndex,
+    		          String details,
+                      LocalDate dueByDateStart,
+                      LocalTime dueByTimeStart,
+                      LocalDate dueByDateEnd,
+                      LocalTime dueByTimeEnd,
+                      String priority,
+                      Set<String> tags) throws IllegalValueException {
+        final Set<Tag> tagSet = new HashSet<>();
+        for (String tagName : tags) {
+            tagSet.add(new Tag(tagName));
+        }
+        this.toEditWith = new Task(
+                new Detail(details),
+                new DueByDate (dueByDateStart, dueByDateEnd),
+                new DueByTime(dueByTimeStart, dueByTimeEnd),
+                new Priority(priority),
+                new UniqueTagList(tagSet)
+        );
+        this.targetIndex = targetIndex;
+        this.isUndoable = true;
+        floating = false;//since if range constructor is used, user would have keyed in a timing 
+        this.removePriority = removePriority; 
+    }
+    
+    public ReadOnlyTask getTask() {
+        return toEditWith;
+    }
+
+```
+###### /java/seedu/cmdo/logic/commands/EditCommand.java
+``` java
+    @Override
+    public CommandResult execute() {
```

Comment:
SLAP can be improved here  ~ i think you've emailed me regarding this ~ yupp hope to see the changes reflected :)